builder/vsphere/supervisor/step_validate_publish.go

```diff
@@ -0,0 +1,134 @@
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type ValidatePublishConfig
+
+package supervisor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	imgregv1a1 "github.com/vmware-tanzu/image-registry-operator-api/api/v1alpha1"
+	vmopv1alpha1 "github.com/vmware-tanzu/vm-operator/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	StateKeyPublishLocationName = "publish_location_name"
+
+	vmPubFeatureNotEnabledMsg = "WCP_VM_Image_Registry feature not enabled"
+)
+
+type ValidatePublishConfig struct {
+	// Name of the location (e.g. vSphere content library) where the source VM will be published.
+	PublishLocationName string `mapstructure:"publish_location_name"`
+}
+
+func (v *ValidatePublishConfig) Prepare() []error {
+	return nil
+}
+
+type StepValidatePublish struct {
+	Config *ValidatePublishConfig
+
+	Namespace  string
+	KubeClient client.Client
+}
+
+func (s *StepValidatePublish) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
```

Comment:
Yes, because we wanted to run this validation as early as possible to avoid creating and watching a source VM to be ready (which can be a lengthy process), in case there is a typo in the specified published location name or it's not writeable. Currently, this step is run after connecting to Supervisor, which is necessary for validating the published location.